CMSIS/Core/Include/m-profile/cmsis_iccarm_m.h

```diff
@@ -128,8 +130,10 @@
 #endif
 
 #ifndef   __NO_RETURN
-  #if __ICCARM_V8
-    #define __NO_RETURN __attribute__((__noreturn__))
+  #if defined(__cplusplus) && __cplusplus >= 201103L
+    #define __NO_RETURN [[noreturn]]
+  #elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+    #define __NO_RETURN _Noreturn
```

Comment:
`_Noreturn` got deprecated in C23 in order to be synced with C++11's `[[noreturn]]`.
https://en.cppreference.com/w/c/language/_Noreturn